client/src/components/related/ProductCard.jsx

```diff
@@ -129,6 +130,22 @@ class ProductCard extends React.Component {
     }
   }
 
+  calcAvgRating(ratings) {
```

Comment:
There are now 3 of us calculating this separately...

One thing I started on but have held off as it is optional and a big change, is to have a more comprehensive 'model' that is bound to the data cache in a shared object/data structure. Such a place would be a natural location to have a shared method for calculating the average ratings from rating data associated with a given product ID.